### ISSUE TITLE
before:browser:launch not running when using --spec

### ISSUE THREAD

**Issue by zxti** [cypress-io/cypress#6674]
### Current behavior:

Running `cypress run --spec SOMESPEC` doesn't run your `before:browser:launch` hooks, only `cypress run` without specific specs does.

### Desired behavior:

Running `cypress run --spec SOMESPEC` runs your `before:browser:launch` hooks.

### Versions

Cypress 4.0.2, tested on Mac and Linux.

**Comment by jennifer-shehane** [comment]
I was unable to confirm the behavior you're describing. 

Ran the code below:

`plugins/index.js

```js
module.exports = (on, config) => {
  on('before:browser:launch', (browser = {}, launchOptions) => {
    console.log('before:browser:launch browser', browser) 
    console.log('before:browser:launch launcOptions', launchOptions)
  })
}
```

Logs the Objects in the STDOUT during `cypress run` with `--spec` flag.

<img width="818" alt="Screen Shot 2020-03-10 at 12 23 48 PM" src="https://user-images.githubusercontent.com/1271364/76283864-257baf00-62ca-11ea-84f8-2cc82000c5b5.png">

Please comment in this issue with a reproducible example and we will consider reopening the issue.

**Comment by gowrishyadalam** [comment]
@jennifer-shehane , I am facing the same issue even i tired with 4.9.0 version. Below is my code
```

  on('before:browser:launch', (browser = {}, launchOptions) => {
    const downloadDirectory = path.join(__dirname, '..', 'mediadownloads')
    if (browser.family === 'chrome') {
        launchOptions.args.push('--window-size=1400,1200')
        launchOptions.preferences.default['download'] = { default_directory: downloadDirectory }
        console.log('before:browser:launch browser', browser)
        console.log('before:browser:launch launcOptions', launchOptions)
        return launchOptions
    }
  })
```
my output is 
====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:    4.9.0                                                                              │
  │ Browser:    Chrome 83                                                                          │
  │ Specs:      14 found (CataloguePage/cataloguePage.js, CompanyPage/companyPage.js, CreateCart/c │
  │             reateCart.js, EquipmentDetailPage/equipmentdetailpage.js, EquipmentListPage/equipm │
  │             entListPage.js, ForgotPasswordPage/forgotPasswordPage.js, InventoryPage/createSupp │
  │             lyModal.js,...)                                                                    │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────

**Comment by jennifer-shehane** [comment]
The `browser.family` should be `chromium` to match Chrome browsers. This changed in 4.0. 

```
if (browser.family === 'chromium') {
```

**Comment by gowrishyadalam** [comment]
Thanks, After changing to chromium it works fine for me

**Comment by gowrishyadalam** [comment]
in before browser launch I am changing the default download directory path, Below is the code

```
  on('before:browser:launch', (browser, launchOptions) => {
    let downloadDirectory = path.join(__dirname, '..', 'mediadownloads')
    if (browser.family === 'chromium') {
          launchOptions.preferences.default['download'] = { default_directory: downloadDirectory }
          return launchOptions
    }
  })
```
after launching the browser default folder "mediadownloads" is not created under the cypress, this use to work for me previously but not now is it expected behavior ??

**Comment by mmbarness** [comment]
I've run into a very similar issue - not sure if I post this in a new issue. At any rate, can't for the life of me get 'before:browser:launch' to trigger, while 'before:run',  'before:spec, 'after:spec', and 'after:run' work just fine. the code looks identical, here it is: 
`on('before:run', details => {
    console.log('before:run')
  })

  on('before:browser:launch', (browser = {}, launchOptions) => {
    console.log('before:browser:launch')
  })

  on('after:run', (results) => {
    console.log('after:run')
  })

  on('before:spec', (spec) => {
    console.log('before:spec')
  })

  on('after:spec', (spec, results) => {
    console.log('after:spec')
  })`

output: 

> cypress run --env fileConfig=staging1 "--headless" "--browser" "chrome" "--spec" "cypress/integration/Test Cases/Tier2/C12_verifyPrequalPunctuation.js"


====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:      8.6.0                                                                            │
  │ Browser:      Chrome 95 (headless)                                                             │
  │ Specs:        1 found (Test Cases/Tier2/C12_verifyPrequalPunctuation.js)                       │
  │ Searched:     cypress/integration/Test Cases/Tier2/C12_verifyPrequalPunctuation.js             │
  │ Experiments:  experimentalInteractiveRunEvents=true                                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘

before:run

────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  Test Cases/Tier2/C12_verifyPrequalPunctuation.js                                (1 of 1)
before:spec
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating


  C12 - Prequal process tolerates extraneous punctuation in form fields
    ✓ C12 doing things (40691ms)


  1 passing (41s)

after:spec

  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     41 seconds                                                                       │
  │ Spec Ran:     Test Cases/Tier2/C12_verifyPrequalPunctuation.js                                 │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘

after:run

====================================================================================================

**Comment by wkillerud** [comment]
Depending on the rest of your plugin file, we may have run into the same issue @mmbarness. I use a third-party plugin that registers `on('before:browser:launch')`, and I register that plugin last. Turns out each event can only have one callback, so my custom handler that was registered before was overwritten. I made an issue for that if it turns out you have the same need: #19809
